### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by fwbrasil**
Hey @williamboxhall, I've started to work on the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,101 @@
-![Clump](clump.png)Clump
+![Clump](clump.png) Clump
+=========================
+Library to ease batching of requests to remote services
```
Comment (fwbrasil) [inline README.md]: I would like to have something shorter.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,101 @@
...
+			new EnrichedTrack(track, userMap(track.user))
+		}
+	}
+}
+```
+
+This example has only one level of nested resources. In a complex system, it is common to have several levels:
+
+```
+timeline
```
Comment (fwbrasil) [inline README.md]: I'm not sure we need this example. I would like to keep the introduction as short as possible. Wdyt?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,101 @@
...
+
+## The solution ##
+
+The complexity comes mainly from declaring together **what** needs to be fetched and **how** it should be fetched. Clump offers an embedded Domain Specific Language (DSL) that allows to declare **what** needs to be fetched and an execution model that determine **how** the resources should be fetched. This execution model has some optimizations to perform requests concurrently, use batches when possible and avoid fetching the same resource multiple times by using a cache.
+
+The DSL is implemented using a monadic interface similar to ```Future```. It is a Free Monad, that only produces a nested series of transformations without triggering the actual execution. This is the characteristic that allows to build the execution model separately from the definition of what needs to be fetched.
+
+The execution model leverages on Applicative Functors, that allows to express independence of computations. It exposes only ```join``` to the user but makes heavy use of other applicative operations internally. This means that even without the user specifying what is independent, the execution model is capable to apply the optimizations.
+
```
Comment (fwbrasil) [inline README.md]: It is necessary to find good sources for these functional concepts and add links.

Comment (fwbrasil) [inline README.md]: Maybe @folone can help. :)

File: README.md (hunk 1 above)
Comment (williamboxhall) [inline README.md]: I liked Jake Donham's tagline "Library for service composition and aggregation"

**Comment by williamboxhall** [comment]
Thanks for starting this, I was hoping to get to it but have just been too busy. Will review this slowly as I can

**Comment by fwbrasil** [comment]
@williamboxhall Take your time, enjoy the holidays. :)

**Comment by fwbrasil** [comment]
@williamboxhall I've just pushed the docs for usage. It is almost done. :)

**Comment by fwbrasil** [comment]
@williamboxhall The text is ready for review.

**Comment by fwbrasil** [comment]
I think you will like the "Internals" section.

**Comment by fwbrasil** [comment]
I'll merge this for now, since the current readme has nothing. Lets iterate on the document to fix the errors on master or new PRs.

**Pull request merged**